modules/algorithm/puzzle_15_bfs.py

```diff
@@ -0,0 +1,75 @@
+class State:
+    def __init__(self, initial_state):
+        self.initial_state = tuple(initial_state)
+
+    def __eq__(self, other):
+        return self.initial_state == other.initial_state
+    
+    def __hash__(self):
+        return hash(tuple(self.initial_state))
```

Comment:
Since your state is a tuple from your constructor you don’t need to recast as a tuple here!